openupgradelib/openupgrade.py

```diff
@@ -3458,6 +3466,75 @@ def convert_to_company_dependent(
         )
 
 
+def convert_from_company_dependent(
+    env,
+    model_name,
+    origin_field_name,
+    destination_field_name,
+    model_table_name=None,
+):
+    """
+    Move a company-dependent field back to the model table.
+
+    The usual setting is: A model didn't have a company_id field in version
+    (n-1), but got company-aware in version (n). Then company-dependent fields
+    don't make sense, and are replaced with plain database columns.
+
+    You're responsible for duplicating records for all companies in whatever way
+    makes sense for the model before calling this function.
+
+    Either call in post-migration or create destination_field_name manually.
+
+    :param model_name: Name of the model.
+    :param origin_field_name: Name of the company-dependent field
+    :param destination_field_name: Name of plain field
+    :param model_table_name: Name of the table. Optional. If not provided
+      the table name is taken from the model (so the model must be
+      registered previously).
+    """
+    table_name = model_table_name or env[model_name]._table
+
+    env.cr.execute(
+        "SELECT id, ttype FROM ir_model_fields "
+        f"WHERE model='{model_name}' AND name='{origin_field_name}'"
+    )
+    if not env.cr.rowcount:
+        logger.warning(
+            "Company-dependent field %s does not exist on model %s",
+            origin_field_name,
+            model_name,
+        )
+        return
+
+    field_id, field_type = env.cr.fetchone()
+
+    value_expression = ir_property_TYPE2FIELD.get(field_type)
+    if value_expression == "value_reference":
+        value_expression = "SPLIT_PART(ip.value_reference, ',', 2)::integer"
+
+    return logged_query(
+        env.cr,
+        f"""
+        UPDATE {table_name}
+        SET {destination_field_name} = (
+            SELECT {value_expression}
+            FROM ir_property ip
+            WHERE ip.fields_id={field_id} --- {origin_field_name}
+            AND (
+                ip.res_id = '{model_name}.' || {table_name}.id
```

Comment:
I insist that this is the problem. The duplicated record has an ID that is not on `ir_property` table, as the corresponding one is the previous record ID.

Reply:
thanks for insisting, see below